### ISSUE TITLE
Limited label problem and startified sampling

### ISSUE THREAD

**Issue by alishibli97** [yurujaja/pangaea-bench#82]
It turns out the code with `limited_label` in the train config doesn't really work at the moment. The indexes used to split the train dataset now are using the Subset util from torch.utils.data. When doing this step (in [line 154](https://github.com/yurujaja/pangaea-bench/blob/1ef146dc96ec80f5baaa5f5a5fb46b861d05f9ff/pangaea/run.py#L154)) the dataset is reinitialized losing some attributes (like classes, mean, std, etc.) from the hydra instantiate function (in [line 142](https://github.com/yurujaja/pangaea-bench/blob/1ef146dc96ec80f5baaa5f5a5fb46b861d05f9ff/pangaea/run.py#L142) for example). As far as I see unfortunately Subset doesn’t integrate well with Hydra instantiation. Is it possible to resinstantiate the dataset like was doing with hydra? Or should we add splitter within every dataset class? I can't seem to see that the first solution is possible, so I am thinking of second solution.

**Comment by alishibli97** [comment]
Hi, thanks for the suggestion! I implemented your idea by creating a GeoFMSubset class that inherits from `torch.utils.data.Subset`, and it works. I'll push the changes to the branch soon. Thanks again!
